cub/test/catch2_test_device_reduce_large_offsets.cu

```diff
@@ -0,0 +1,244 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "insert_nested_NVTX_range_guard.h"
+// above header needs to be included first
+
+#include <cub/device/device_reduce.cuh>
+#include <cub/thread/thread_operators.cuh>
+
+#include <thrust/iterator/counting_iterator.h>
+
+#include <cstdint>
+
+#include "catch2_test_device_reduce.cuh"
+#include "catch2_test_helper.h"
+#include "catch2_test_launch_helper.h"
+
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Reduce, device_reduce);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Sum, device_sum);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Min, device_min);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMin, device_arg_min);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::Max, device_max);
+DECLARE_LAUNCH_WRAPPER(cub::DeviceReduce::ArgMax, device_arg_max);
+
+// %PARAM% TEST_LAUNCH lid 0:1:2
+
+// List of offset types to test
+using offset_types = c2h::type_list<std::int32_t, std::uint32_t, std::uint64_t>;
+
+__host__ __device__ __forceinline__ uint64_t
+get_segmented_guassian_sum(const uint64_t num_items, const uint64_t segment_size)
+{
+  uint64_t sum_per_full_segment = (segment_size * (segment_size - 1)) / 2;
+  uint64_t full_segments        = num_items / segment_size;
+  uint64_t index_within_segment = num_items % segment_size;
+
+  uint64_t sum_within_partial_segment = (index_within_segment * (index_within_segment - 1)) / 2;
+  uint64_t sum_over_full_segments     = full_segments * sum_per_full_segment;
+  return sum_within_partial_segment + sum_over_full_segments;
+}
+
+template <typename ItemT>
+struct mod_op
+{
+  uint64_t segment_size;
+
+  __host__ __device__ __forceinline__ uint64_t operator()(const uint64_t index) const
```

Comment:
General question: why are we not using `_CCCL_HOST_DEVICE _CCCL_FORCEINLINE` here?